AK/FlyString.cpp

```diff
@@ -137,7 +96,7 @@ bool FlyString::operator==(FlyString const& other) const
 
 bool FlyString::operator==(String const& other) const
 {
-    if (m_data == other.to_fly_string_data({}))
+    if (m_data == other)
```

Comment:
This (and the unchanged `m_data == other.m_data` in `FlyString::operator==(FlyString const&)`) undermines the entire point of FlyString, as comparison is no longer constant-time (rather, it's equivalent to string comparison because of <https://github.com/SerenityOS/serenity/pull/21661/commits/f88b22fe944078df9cc1fa5eef249e14887fcb78#diff-640ab46988fb013ce6d02bd7aa8da1f13ae6a4ddd89b74855cb85e8847e044a6R49-R52>).

You need to either:
- Make the hashmap hold `NNOP<StringData>` instead of just `StringData`, and make this class hold only a reference to that
- Compare the underlying buffer ptr + length of the string data here and wherever flystring comparison semantics are needed.

Reply:
I can just compare `m_data` as pointer if both of the `StringBase`s are marked as fly strings.